src/sdl2_ui.cpp

```diff
@@ -773,7 +773,7 @@ void Sdl2Ui::ProcessFingerEvent(SDL_Event& evnt) {
 }
 
 void Sdl2Ui::SetAppIcon() {
-#if defined(__WINRT__) || defined(__MORPHOS__)
+#if defined(__WINRT__) || defined(__MORPHOS__) || TARGET_OS_OSX
```

Comment:
Added this because on macOS, running the code in this function causes the app icon in the Dock to become really blurry.

Reply:
Sure, because it gets replaced by a 32x32 icon. No idea of the exact strategy macos uses, but I assume it uses the icon of the app bundle otherwise.
However, this does not apply to Homebrew for example.

Reply:
Good point about homebrew! I could make it specifically check if it's running in an app bundle and then early exit?

Reply:
Seems like a good idea. This way we still have an icon when not run as app bundle.

---
General problem is the SDL way of setting this is limited, so I already added a windows specific way.
With high resolutions / HighDPI desktops this needs to be addressed sooner or later.
We already have high resolution variants of our icon, problem is we do not ship them with the raw executable, because it would increase the size. So a 32x32 icon was added to not make the executable much bigger.
On Linux-like platforms we could use the freedesktop.org icon specification to get the svg (or scaled pngs), for macos (with homebrew) we could put the `.icns` somewhere, for Windows use the included `.ico`.
However, this is unrelated to this Pull Request.